docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1061 @@
+![AddCommandSequenceDiagramReference](static/frontend/commands/AddCommandSequenceDiagramReference.png)
+
+Given below is an example usage scenario and how the add mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `EntryLog` will be initialized and contains no entry.
+
+Step 2. The user executes `/add Lunch at McDonalds -category Food -price 19.9` command to add an `Entry` to the `EntryLog`.
+
+_***Note.*** The command will fail its execution if its format is incorrect, and no `Entry` will be added to the `Entrylog`. An error message will be displayed informing the user._
+
+Step 3. The command will be resolved by `Parser`, which would create an `AddCommmand` object.
+
+Step 4. The `AddCommand` constructor creates and returns an `Entry` object containing the description, price and category to be added.
+
+Step 5. When `execute()` method is called, a `Request` object is created.
+
+Step 6. From there, the `Request` is ready to be handled. `addEntry()` method is called and the new `Entry` is added to the `EntryLog`.
+
+Step 7. A success message is after the new `Entry` is added to the `EntryLog`.
+
+The following activity diagram summarizes what happens when a user executes an add command:
+
+![AddCommandActivityDiagram](static/frontend/commands/AddCommandActivityDiagram.png)
+
+#### Delete Command
+The 'delete' entry mechanism is facilitated by `EntryLog`.
+
+Every instance of `DeleteCommand` is created with an Integer, which is the ID of the `Entry` to be deleted.
+
+The following sequence diagram shows how the delete command work:
+
+![DeleteCommandSequenceDiagram](static/frontend/commands/DeleteCommandSequenceDiagram.png)
+![DeleteCommandSequenceDiagramGetResponse](static/frontend/commands/DeleteCommandSequenceDiagramDeleteResponse.png)
+![DeleteCommandSequenceDiagramDeleteResponse](static/frontend/commands/DeleteCommandSequenceDiagramDeleteResponse.png)
+
+Given below is an example usage scenario and how the delete mechanism behaves at each step.
+
+Step 1. The user decides to remove an `Entry` from the `EntryLog` and executes `/delete 1` command.
+
+_**Note:** The command will fail its execution if the index provided is invalid, and no `Entry` will be removed from the `EntryLog`. An error message will be displayed informing the user._
+
+Step 2. The command will be resolved by `Parser`, which would create an `DeleteCommmand` object containing the index of the `Entry` to be deleted.
+
+Step 3. When `execute()` method is called, a `Request` object is created.
+
+Step 4. From there, the `Request` is ready to be handled. `deleteEntry()` method is called and the `Entry` is removed from `EntryLog`.
+
+Step 5. A success message is after the `Entry` is removed from `EntryLog`.
+
+The following activity diagram summarizes what happens when a user executes a delete   command:
+
+![DeleteCommandActivityDiagram](static/frontend/commands/DeleteCommandActivityDiagram.png)
```

Comment:
Such summary diagrams are useful in giving the reader a good summary of what occurs when the command is executed. I would recommend using these diagrams more consistently and placing such diagrams right before you elaborate before going into deeper detail about what happens in each step. This makes it more readable.